trainingFunction.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+function dbPull():array{
+    $connectionString = 'mysql:host=db; dbname=training_diary';
+    $dbUsername = 'root';
+    $dbpassword = 'password';
+    $db = new PDO ($connectionString, $dbUsername, $dbpassword);
+
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+    $queryString = 'SELECT * FROM 
+             `training_log`
+                 ORDER BY `date` DESC 
+                 LIMIT 10;'; //order by date descending
+    $query = $db->prepare($queryString);
+    $query->execute();
+    return $query->fetchAll();
+}
+
+function addHtmlToWorkouts(array $all_workouts_array):array{
+    foreach ($all_workouts_array as $key=>$workout_item){
+        if (!is_array($workout_item) ||
+            !array_key_exists('date', $workout_item) ||
+            !array_key_exists('exercise', $workout_item) ||
+            !array_key_exists('weight_added_kg', $workout_item) ||
+            !array_key_exists('comments', $workout_item)) {
+            return [];
+        }
+        $exercise = $workout_item['exercise'];
+        $weight_added_kg = ($workout_item['weight_added_kg']===null) ? 0 : $workout_item['weight_added_kg'];
+        $comments = ($workout_item['comments']===null) ? '' : $workout_item['comments'];
+        $all_workouts_array[$key]['html'] = "<div class='exercise_container'><h3> $exercise <span class='weight_added_span'> | " . $weight_added_kg . "kg Added </span></h3><p>$comments</p></div>";
+    }
+    return $all_workouts_array;
+}
+
+function perDateOutput(array $all_workouts_array, array $date_array):string{
```

Comment:
i would rename this function to something like `displayAllWorkouts` because that would make it easier to understand what it does at a glance